Darwin: promote Cooking Appliances to Introduced

kiel-apple opened the pull request:
Continuing the theme of v1.3 promotions.  Added comments about `WATTS` and affected provisional bits.

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -8823,6 +8827,52 @@
+              - MinPower
+              - MaxPower
+              - PowerStep
+              # SupportedWatts and SelectedWattIndex are Provisional in Matter 1.3
```
Then they need explicit provisional annotations, right?

kiel-apple replied:
Yep - is [this](https://github.com/project-chip/connectedhomeip/pull/33399/files#diff-94ddfb24d1680d06578d1e18f05664162d7c7e9f05958813917cde261c7e9541R9765-R9767) correct?

bzbarsky-apple replied:
Generally, yes, but I would add a comment explaining why those are marked provisional.

andy31415 commented:
@kiel-apple this PR seems quite old and has merge conflicts. Is it supposed to go in?

kiel-apple commented:
@andy31415 let me take a stab at bringing it back up to date today, I think it won't be awful.

kiel-apple committed:
cooking appliances: add MicrowaveOven + Oven clusters

and codegen

add Microwave + Oven attributes

and codegen

more Microwave + Oven attributes

and codegen

add Microwave + Oven commands

and codegen

add Microwave + Oven command fields

and codegen

add Microwave + Oven structs

and codegen

add Microwave + Oven struct fields

and codegen

add Microwave + Oven events/event fields

and codegen

add Microwave + Oven enums

and codegen

add Microwave + Oven enum values

and codegen

add Microwave + Oven bitmaps / bitmap values

and codegen

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9685,13 +9685,156 @@
+          MicrowaveOvenMode:
+              - SupportedModes
+              - CurrentMode
+              - StartUpMode
+              - OnMode
```
These are marked X in the spec, no?  We shouldn't have them here.

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9685,13 +9685,156 @@
+          OvenMode:
+              - SupportedModes
+              - CurrentMode
+              - StartUpMode
+              - OnMode
```
Again, StartUpMode and OnMode are X in the spec for Oven Mode.

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9685,13 +9685,156 @@
+              - SetCookingParameters
+              - AddMoreTime
+          MicrowaveOvenMode:
+              - ChangeToMode
+              - ChangeToModeResponse
```
These are set as X in the spec, no?

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9685,13 +9685,156 @@
+              - ChangeToMode
+              - ChangeToModeResponse
+          OvenCavityOperationalState:
+              - Pause
+              - Stop
+              - Start
+              - Resume
```
Pause and Resume are X in the spec.

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9742,6 +9918,18 @@
+                  - PowerNumberLimits
+          MicrowaveOvenMode:
+              Feature:
+                  - OnOff
```
This feature is disallowed for this cluster, no?

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9742,6 +9918,18 @@
+                  - OnOff
+          OvenMode:
+              Feature:
+                  - OnOff
```
Likewise, this is disallowed....

bzbarsky-apple reviewed (changes requested):
I would recommend removing the provisional annotations for things being added; that would also make it clearer what's still marked as provisional.

kiel-apple committed:
Remove Microwave + Oven X attributes and commands

kiel-apple committed:
remove Microwave + Oven OnOff

is X conformance

kiel-apple committed:
codegen for Microwave + Oven

kiel-apple commented:
@bzbarsky-apple thanks for checking the spec, I could have done better there given how long it has been since I opened this

kiel-apple commented:
> I would recommend removing the provisional annotations for things being added; that would also make it clearer what's still marked as provisional.

I still need to go back and do this.  Coming soon!

kiel-apple committed:
further clarify provisional status of provisional WATTS items

kiel-apple committed:
remove old Microwave + Oven provisional cluster entries from Spring 2024

kiel-apple commented:
need to move all these entries forward to new Future since Future from back when this started has been released

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -9689,13 +9685,144 @@
   introduced:
+      clusters:
+          - MicrowaveOvenControl
+          - MicrowaveOvenMode
```
So... now when I run codegen with this PR as of right now (rev cd2fc6b3f63aeeb051a6b99b461427609f95fc08) I see a bunch of warnings about thigns that are not annotated as either provisional, removed, or introduced, right?

kiel-apple committed:
move Microwave + Oven changes to new Future stanza

kiel-apple committed:
remove provisional items from old future

kiel-apple committed:
additional tweaks from codegen output

restyled-commits committed:
Restyled by prettier-yaml

kiel-apple committed:
remove Microwave + Oven Feature bitmaps for now

empty enum

kiel-apple committed:
rerun codegen

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -10084,9 +10080,245 @@
           - WaterHeater
+      attributes:
+          MicrowaveOvenControl:
+              - SupportedWatts
+              - SelectedWattIndex
```
The comment about why these are provisional should be here.

kiel-apple replied:
added.

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -10084,9 +10080,245 @@
+              - MinPower
+              - MaxPower
+              - PowerStep
+              # SupportedWatts and SelectedWattIndex are Provisional because WATTS is Provisional in Matter 1.3
```
We're looking at Matter 1.4 now, no?  Please fix comments to reflect 1.4 status?

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -10084,9 +10080,245 @@
+              - SetCookingParameters
+              - AddMoreTime
+          MicrowaveOvenMode:
+              - ChangeToMode
+              - ChangeToModeResponse
```
These are X in the spec, and presumably do not exist?  Why are they here?

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -10084,9 +10080,245 @@
+  removed:
+      attributes:
+          OvenMode:
+              - StartUpMode
+              - OnMode
```
Please document why these are "removed" even though they are still in the XML?

And please file an issue to get the XML fixed.  Same for OvenCavityOperationalState.

bzbarsky-apple commented on src/darwin/Framework/CHIP/templates/availability.yaml:
```diff
@@ -10084,9 +10080,245 @@
+              - Feature
+          OvenMode:
+              - Feature
+      bitmap values:
```
Why do you need these?  if the bitmap is removed, then the bits inside it are removed automatically....  I guess the bitmap values removals got added first to implement the X from the spec, and then that failed to compile and the bitmap removals above were added?

In any case, either just remove the bitmaps or clearly document why the redundant bits are here (e.g. so that if the bitmaps ever get un-removed due to growing new bits we will not get incorrect bits in them).

kiel-apple committed:
additional / more accurate context for WATTS entries

kiel-apple commented:
incorporated into #36793
